azurerm/internal/services/netapp/resource_arm_netapp_volume.go

```diff
@@ -89,6 +90,19 @@ func resourceArmNetAppVolume() *schema.Resource {
 				ValidateFunc: azure.ValidateResourceID,
 			},
 
+			"protocols": {
+				Type:     schema.TypeList,
+				ForceNew: true,
+				Optional: true,
+				MaxItems: 2,
```

Comment:
You can only select two protocols? or do two conflict?

Reply:
Only one combination of two values will work, cifs+nfsv3 or cifs+nfsv4.1, three items implies that the user is adding nfsv3 or nfvs4.1 together and this is a combination that fails. I want to write a validator function to check if nfsv3 and nfsv4 are at the same time defined in the list but it looks like the validation is done on item by item, if you have an example where we can validade if two items are in the list that would be great, otherwise we can skip to a next update since this may take more time to be done.

Reply:
You can add a validation function to the list block itself:
```
func ListOfUniqueStrings(i interface{}, k string) (warnings []string, errors []error) {
	v, ok := i.([]interface{})
	if !ok {
		errors = append(errors, fmt.Errorf("expected type of %q to be List", k))
		return warnings, errors
	}

	for _, e := range v {
		if _, eok := e.(string); !eok {
			errors = append(errors, fmt.Errorf("expected %q to only contain string elements, found :%v", k, e))
			return warnings, errors
		}
	}

	for n1, i1 := range v {
		for n2, i2 := range v {
			if i1.(string) == i2.(string) && n1 != n2 {
				errors = append(errors, fmt.Errorf("expected %q to not have duplicates: found 2 or more of %v", k, i1))
				return warnings, errors
			}
		}
	}

	return warnings, errors
}
```

Reply:
i'd also document the restriction

Reply:
I'm specifying single value for now on doc support for multi-protocol is on the road map but not public ETA. With this in mind we can define maxitem=1 on both for now, what do you think?

Reply:
👍 sounds good

Reply:
> I think we'll want to add ForceNew to this as well. As `ProtocolTypesCanNotBeChanged`

@mbfrahry, not sure I understood what is needed here, do you have more details on the "As `ProtocolTypesCanNotBeChanged`" sentence?